src/gui/folderwatcher.cpp

```diff
@@ -190,4 +206,62 @@ void FolderWatcher::changeDetected(const QStringList &paths)
     }
 }
 
+void FolderWatcher::possiblyAddUnlockedOfficeFile(const QString &path, QSet<QString> &container)
+{
+    qCInfo(lcFolderWatcher) << "Checking if it is an office lock file:" << path;
+    QString officeLockFilePrefixFound;
+    for (const auto &currentPrefix : officeLockPrefixes) {
+        if (path.contains(currentPrefix)) {
+            officeLockFilePrefixFound = currentPrefix;
+            break;
+        }
+    }
+
+    if (officeLockFilePrefixFound.isEmpty() || QFileInfo::exists(path)) {
+        return;
+    }
+
+    qCInfo(lcFolderWatcher) << "Found an office lock file with prefix:" << officeLockFilePrefixFound << "in path:" << path;
+
+    const auto lockFilePathWitoutPrefix = QString(path).replace(officeLockFilePrefixFound, QStringLiteral(""));
+    auto lockFilePathWitoutPrefixSplit = lockFilePathWitoutPrefix.split(QLatin1Char('.'));
+
+    if (lockFilePathWitoutPrefixSplit.size() < 2) {
+        return;
+    }
+
+    const auto lockFileExtensionDirty = lockFilePathWitoutPrefixSplit.takeLast();
+    // remove possible non-alphabetical characters at the end of the extension
+    QString extensionSanitized;
+    for (const auto &ch : lockFileExtensionDirty) {
+        if (ch.isLetter()) {
+            extensionSanitized += ch;
+        }
+    }
+    lockFilePathWitoutPrefixSplit.push_back(extensionSanitized);
+    auto unlockedFilePath = lockFilePathWitoutPrefixSplit.join(QLatin1Char('.'));
```

Comment:
maybe split that into another function that would be called `sanitizePath`
seems pretty specific to a specific software